src/search/json_index/json_search.ml

```diff
@@ -192,11 +206,26 @@ let output_json ppf first entries =
           true)
     first entries
 
-let unit ppf u =
+let unit ?occurrences ppf u =
+  let get_occ id =
+    match occurrences with
+    | None -> None
+    | Some occurrences -> (
+        (* We don't want to include the [sub] field of occurrence tables. We use
+           a "polymorphic record" to avoid defining a type, but still get named
+           fields! *)
```

Comment:
This is original but I find it confusing and heavy in syntax. What about defining a record with just that instead ?

This record could even replace the `item` type in `Table` as the `sub` field is never used outside of this module.
Here's an other idea: https://github.com/Julow/odoc/commit/9cafc01c8f803828f15923dca18020010da43924

Reply:
I was thinking that the `sub` field could be useful to some application of the occurrence table.

However, there are no such application. I removed `sub` from the exposed type in the last commit, and use that type instead of this weird tuple. If an application requires exposing the `sub` field, we'll do it at this moment.